master/src/main/java/org/apache/celeborn/service/deploy/master/clustermeta/AbstractMetaManager.java

```diff
@@ -284,27 +289,49 @@ public void updateWorkerHeartbeatMeta(
     }
 
     // If using HDFSONLY mode, workers with empty disks should not be put into excluded worker list.
-    long unhealthyDiskNum =
-        disks.values().stream().filter(s -> !s.status().equals(DiskStatus.HEALTHY)).count();
-    boolean exceed = unhealthyDiskNum * 1.0 / disks.size() >= unhealthyDiskRatioThreshold;
+    Pair<Boolean, Long> exceedCheckResult = isExceedingUnhealthyThreshold(disks);
     if (!excludedWorkers.contains(worker)
-        && (((disks.isEmpty() || exceed)
-                && !conf.hasHDFSStorage()
-                && !conf.hasS3Storage()
-                && !conf.hasOssStorage())
+        && (((disks.isEmpty() || exceedCheckResult.getLeft()) && !hasRemoteStorage())
```

Comment:
let's use a method `hasStorage()` for this `((disks.isEmpty() || exceedCheckResult.getLeft()) && !hasRemoteStorage())` eg:`!hasStorage()`